docusaurus/docs/iOS/customization/localization.md

```diff
@@ -0,0 +1,138 @@
+---
+title: Localization
+---
+
+StreamChat offers pre-defined localization for text indicating across the whole UISDK. Currently, StreamChat supports only English language, but if you want to provide your custom strings, it's very easy to do.
+The whole approach to StreamChat SDK is to stay as much consistent as possible with native iOS paradigms and approaches to certain problems. That's why we use `strings` and `stringsdict` files to handle localize the framework.
+For advanced plural localization cases (typing indicator showing how many people are typing or how many responses are in thread), we use `stringsdict` with `NSStringPluralRuleType` 
+
+## Adding Custom Localization
+
+As said earlier, StreamChatUI uses `strings` and `stringsdict` files to handle localize the framework.
+`Appearance` structure contains closure `localizationProvider`, which is responsible for deriving `localizedString` from exact bundle. 
+By default, the bundle is  `StreamChatUI` framework bundle, but you can customize it to anything you need in 3 steps:
+
+1. First thing you need to do is to copy `StreamChatUI` localization keys into your `strings` file (Or just copy the default `Localizable.strings` and `Localizable.stringsdict` files to your project from our repository).
+2. Once you have those files in your application, you need to set the `localizationProvider` to provide your `AppBundle` instead of `StreamChatUI` frameworks like this: 
+```swift
+Appearance.default.localizationProvider = { key, table in
```

Comment:
why don't we do this by default?

Reply:
what I mean is that by default we do this:

- Lookup string on Bundle.main.localizedString
- If nothing is found lookup the string from the SDK

this way adding translations to the app bundle will require no other change and you only need to touch `Appearance.default.localizationProvider` if you do something different/less common

Reply:
@tbarbugli that's simply not possible. SDK provides it's own bundle and app provides a totally different bundle. we can't reach app bundle from SDK